### ISSUE TITLE
Upgrade Path to osm-tasking-manager2

### ISSUE THREAD

**Issue by schnuerle** [hotosm/osm-tasking-manager#250]
Is there a good resource for how to upgrade an installation of osm-tasking-manager to osm-tasking-manager2?  

There is some interest at https://lists.openstreetmap.org/pipermail/imports-us to upgrade the US Import installation at http://tasks.openstreetmap.us/ (which is still heavily used) to version 2, but I'm not sure the admins know the best way to do this.

While I have your attention... Does V2 allow for arbitrary polygon areas, like this task? http://tasks.openstreetmap.us/job/50  

Thanks.

**Comment by pgiraud** [comment]
Hey!

I can certainly help with the migration. I did the migration once for HOT. I have a quick and ugly script to automate the migration. https://github.com/pgiraud/osmtm_v1_migration
You will still need my help though since I'm not sure it'll work the first time.
The best option is probably the following one: if there's no private data, an admin can send me the last version of the v1 database. It's as simple as sending me the OSMTM.db file. I'll run the script with this file to see where I get to.

And yes, the V2 can work with arbitrary areas.

**Comment by iandees** [comment]
@pgiraud very cool! I'll send you the sqlite db to the e-mail in your github profile.

**Comment by schnuerle** [comment]
Hey nice stuff you two!  I'll reply back on the [import-us] mailing list so others can see.

**Comment by schnuerle** [comment]
Has this upgrade happened, and if so is there a new link for the US Task Manager?  I'd like to 1) review the progress of our Louisville import and 2) start a new import task.  

The site http://tasks.openstreetmap.us/ now gets a Bad Gateway error.

Thanks.

**Comment by pgiraud** [comment]
As far as I know, the migration never happened.

**Comment by iandees** [comment]
The migration didn't happen and the disk that held the database died. I haven't been able to replace it since. I'm planning on setting it up this week, though.

I imagine we'll have to start with a fresh, empty Task Manager database.

**Comment by schnuerle** [comment]
Thanks for the quick replies.  Sorry to hear about the loss of data and all your troubles!  

I hope some people in the middle of imports weren't affected too much.  Our import was completed in May but I'm going to present it at Code for America's annual summit at the end of the month and wanted to show it off.  But I've got some screen caps anyway.

When you set it up this weekend, can you configure it as osm-tasking-manager2?  No data migration needed!

**Comment by iandees** [comment]
Yep, my plan is to use the new tasking manager code.

**Comment by severinmenard** [comment]
Is it why all the instructions in French for
http://tasks.hotosm.org/project/108 vanished?
I hope there is a way to get this back, considering the hours it takes to
translate.

Sincerely,

Severin

**Comment by iandees** [comment]
No, tasks.hotosm.org has nothing to do with tasks.openstreetmap.us.

**Comment by severinmenard** [comment]
Ah sorry, I did not see it was about the US instance

**Comment by schnuerle** [comment]
Hey Ian, have you made any progress on the install?  The Code for America Summit is in 2 weeks and I'm presenting how local devs can import open data for their city, similar to what we did for Louisville: http://www.civicdataalliance.org/gis-open-data-american-printing-house-open-street-map/  

I'd love to be able to show them the new Task Manager site and have at least one real task setup to demo.  Thanks a lot.

**Comment by iandees** [comment]
I ran into several issues with dependencies and finally got them fixed last
night. I still need to get the WSGI server running and redirected, etc.

On Mon, Sep 14, 2015 at 9:47 AM, Michael Schnuerle <notifications@github.com

**Comment by iandees** [comment]
I think I have it running on tasks.openstreetmap.us. If you're interested,
please give it a shot.
